treecorr/binnedcorr2.py

```diff
@@ -1300,17 +1303,54 @@ def _make_cov_design_matrix(corrs, plist, func, name):
     # the original.
     corrs = [c.copy() for c in corrs]
 
+    # We can't pickle functions to send via MPI, so have to do this here.
+    if func is None:
+        func = lambda corrs: np.concatenate([c.getStat() for c in corrs])
+
+    # We also can't send generators using MPI, so instead we need
+    # to create them here.
+    if name == "jackknife":
+        plist = [c._jackknife_pairs() for c in corrs]
+        # Swap order of plist.  Right now it's a list for each corr of a list for each row.
+        # We want a list by row with a list for each corr.
+        plist = list(zip(*plist))
+    elif name == "sample":
+        plist = [c._sample_pairs() for c in corrs]
+        # Swap order of plist.  Right now it's a list for each corr of a list for each row.
+        # We want a list by row with a list for each corr.
+        plist = list(zip(*plist))
+    elif name == "marked_bootstrap":
+        nboot = np.max([c.num_bootstrap for c in corrs])
+        plist = []
+        for k in range(nboot):
+            # Select a random set of indices to use.  (Will have repeats.)
+            indx = corrs[0].rng.randint(npatch, size=npatch)
+            vpairs = [c._marked_pairs(indx) for c in corrs]
+            plist.append(vpairs)
+    elif name == "bootstrap":
+        nboot = np.max([c.num_bootstrap for c in corrs])  # use the maximum if they differ.
+        plist = []
+        for k in range(nboot):
+            indx = corrs[0].rng.randint(npatch, size=npatch)
+            vpairs = [c._bootstrap_pairs(indx) for c in corrs]
+            plist.append(vpairs)
+    else:
+        raise ValueError("Unknown name %s in _make_cov_design_matrix_core" % name)
```

Comment:
I don't think you can get to this point using the public API.  As you probably noticed by the fact that you had to write the unit test using a private method (something I normally try to avoid).  So this should just be `assert false` with a comment that you can't get here.  And with `pragma: no cover` to keep the coverage checker happy.
Of course, all of this is moot if you can move this work back to the separate cov functions, since this if/else chain wouldn't exist anymore.

Reply:
Removed, and I've taken out the test.